launch/launch/action.py

```diff
@@ -56,7 +56,21 @@ def describe(self) -> Text:
     def visit(self, context: LaunchContext) -> Optional[List[LaunchDescriptionEntity]]:
         """Override visit from LaunchDescriptionEntity so that it executes."""
         if self.__condition is None or self.__condition.evaluate(context):
-            return cast(Optional[List[LaunchDescriptionEntity]], self.execute(context))
+            try:
+                return cast(Optional[List[LaunchDescriptionEntity]], self.execute(context))
+            finally:
+                from .events import ExecutionComplete  # noqa
+                future = self.get_asyncio_future()
```

Comment:
Hmm correct me if I'm wrong, but I was expecting the future, if any, to be always the same and to be resolved by the action. The rationale is, if there are ongoing tasks (i.e. `future is not None`) emit the event once those tasks are done (i.e. `future.done() is True`), otherwise emit right away.

Reply:
You're right, I read `self.get_asyncio_future()` as one of the future _creation_ functions, but it's actually an accessor.